playbooks/common/openshift-cluster/validate_hostnames.yml

```diff
@@ -8,16 +8,27 @@
     register: lookupip
     changed_when: false
     failed_when: false
-  - name: Warn user about bad openshift_hostname values
-    pause:
-      prompt:
+
+  - name: Validate openshift_hostname when defined
+    fail:
+      msg: >
         The hostname {{ openshift.common.hostname }} for {{ ansible_nodename }}
         doesn't resolve to an IP address owned by this host. Please set
         openshift_hostname variable to a hostname that when resolved on the host
         in question resolves to an IP address matching an interface on this
-        host. This host will fail liveness checks for pods utilizing hostPorts,
-        press ENTER to continue or CTRL-C to abort.
-      seconds: "{{ 10 if openshift_override_hostname_check | default(false) | bool else omit }}"
+        host. This host will fail liveness checks for pods utilizing hostPorts.
+        See https://docs.openshift.org/latest/install_config/install/advanced_install.html#configuring-host-variables
     when:
     - lookupip.stdout != '127.0.0.1'
     - lookupip.stdout not in ansible_all_ipv4_addresses
+    - not openshift_override_hostname_check | default(false)
```

Comment:
Suggest to change openshift_override_hostname_check to openshift_hostname_check to reflect this check. And change this line to "openshift_hostname_check | default(true)" to make this check happen by default, or else, later node SDN start would encounter failure if the hostname doesn't resolve to an IP address owned by this host.